src/objects.c

```diff
@@ -874,7 +968,9 @@ void disconnect_server(PgSocket *server, bool notify, const char *reason, ...)
 	free_scram_state(&server->scram_state);
 
 	server->pool->db->connection_count--;
-	server->pool->user->connection_count--;
+	if (server->pool->user) {
```

Comment:
should we add a comment somewhere that user is NULL in case of peer pools, because we only use them to propagate cancellations, which do not have a specific user.

Reply:
mentioned this in the struct definition in `bouncer.h` and also in the function comment of `new_peer_pool`